src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -690,6 +690,9 @@ public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationT
             }
         }
 
+        //PROTOTYPE: hook to do generation, by default does nothing
+        protected virtual Compilation RunGenerators(Compilation input, ParseOptions parseOptions, ImmutableArray<AdditionalText> additionalTexts) { return input; }
```

Comment:
>RunGenerators [](start = 38, length = 13)

It would be good to add more details about the inputs and the output in the form of a doc comment